src/encoder/serde.rs

```diff
@@ -507,6 +507,17 @@ impl Serialize for RegExp {
     }
 }
 
+impl Serialize for JavaScriptCodeWithScope {
```

Comment:
This follows the pattern of the other serializers defined here. Since the serialize implementation for Bson implements serializing from extjson, we piggyback off that to serialize JavaScriptCodeWithScope.